File: backend/app/api/routes/analyze.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.models.encounter import EncounterRequest
from app.models.response import AnalyzeResponse
from app.services.analyzer import analyze_encounter

router = APIRouter(tags=["analyze"])


@router.post("/analyze-encounter", response_model=AnalyzeResponse)
def post_analyze_encounter(request: EncounterRequest, db: Session = Depends(get_db)) -> AnalyzeResponse:
    try:
        return analyze_encounter(request, db)
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"Encounter analysis failed: {exc}") from exc

File: frontend/src/lib/api/analyze.ts
import apiClient from "./client";
import type { EncounterRequest } from "@/types/encounter";
import type { AnalyzeResponse } from "@/types/analysis";

export async function analyzeEncounter(
  payload: EncounterRequest
): Promise<AnalyzeResponse> {
  const { data } = await apiClient.post<AnalyzeResponse>(
    "/analyze-encounter",
    payload
  );
  return data;
}